### PR TITLE
Gitops global config

### PR DESCRIPTION
**Pull request by nishant-d**

### CODE REVIEW

File: pkg/gitops/GitOpsConfigService.go [hunk 1]
```diff
@@ -82,17 +83,34 @@ func NewGitOpsConfigServiceImpl(Logger *zap.SugaredLogger, ciHandler pipeline.Ci
...
+		}
+	}
+
 	model := &repository.GitOpsConfig{
 		Provider:      request.Provider,
 		Username:      request.Username,
 		Token:         request.Token,
 		GitHubOrgId:   request.GitHubOrgId,
 		GitLabGroupId: request.GitLabGroupId,
 		Host:          request.Host,
```
Comment (nishant-d) [inline pkg/gitops/GitOpsConfigService.go]: in this window there won't be any active config

File: internal/util/K8sUtil.go [hunk 2]
```diff
@@ -96,6 +100,37 @@ func (impl K8sUtil) deleteNs(namespace string, client *v12.CoreV1Client) error {
 	return err
 }
 
+func (impl K8sUtil) getargoAppClient(clusterConfig *ClusterConfig) (*rest.RESTClient, error) {
```
Comment (nishant-d) [inline internal/util/K8sUtil.go:103]: whats use of this method?

File: internal/util/K8sUtil.go [hunk 3]
```diff
@@ -96,6 +100,37 @@ func (impl K8sUtil) deleteNs(namespace string, client *v12.CoreV1Client) error {
...
+	config.Host = clusterConfig.Host
+	config.BearerToken = clusterConfig.BearerToken
+	config.Insecure = true
+	config.NegotiatedSerializer = serializer.NewCodecFactory(runtime.NewScheme())
+
+	client, err := rest.RESTClientFor(config)
+	return client, err
+}
+
+func (impl K8sUtil) CreateArgoApplication(namespace string, application string, clusterConfig *ClusterConfig) error {
```
Comment (nishant-d) [inline internal/util/K8sUtil.go:117]: whats use of this method?

File: pkg/gitops/GitOpsConfigService.go [hunk 4]
```diff
@@ -0,0 +1,471 @@
...
+
+	secret, err := impl.K8sUtil.GetSecretFast(impl.aCDAuthConfig.ACDConfigMapNamespace, GitOpsSecretName, client)
+	statusError, _ := err.(*errors.StatusError)
+	if err != nil && statusError.Status().Code != http.StatusNotFound {
+		impl.logger.Errorw("secret not found", "err", err)
+		return nil, err
+	}
+	if secret == nil {
+		data := make(map[string][]byte)
+		data["username"] = []byte(request.Username)
```
Comment (nishant-d) [inline pkg/gitops/GitOpsConfigService.go:155]: base64 encoding is not required here?

File: pkg/gitops/GitOpsConfigService.go (hunk 1 above)
Comment (vikramdevtron) [reply pkg/gitops/GitOpsConfigService.go]: yes if user forcefully deactivate.

File: internal/util/K8sUtil.go (hunk 2 above)
Comment (vikramdevtron) [reply internal/util/K8sUtil.go:103]: was used for create acd apps, not in use now.

File: internal/util/K8sUtil.go (hunk 3 above)
Comment (vikramdevtron) [reply internal/util/K8sUtil.go:117]: not in use

File: pkg/gitops/GitOpsConfigService.go (hunk 4 above)
Comment (vikramdevtron) [reply pkg/gitops/GitOpsConfigService.go:155]: for username or for both username and token ?

Comment (vikramdevtron) [reply pkg/gitops/GitOpsConfigService.go:155]: > base64 encoding is not required here?

for username or for both username and token ?

File: internal/util/K8sUtil.go (hunk 3 above)
Comment (vikramdevtron) [reply internal/util/K8sUtil.go:117]: > whats use of this method?

not in use now.

File: internal/util/K8sUtil.go (hunk 2 above)
Comment (nishant-d) [reply internal/util/K8sUtil.go:103]: then please remove it.

File: internal/util/K8sUtil.go (hunk 3 above)
Comment (nishant-d) [reply internal/util/K8sUtil.go:117]: then please remove it

File: pkg/gitops/GitOpsConfigService.go (hunk 4 above)
Comment (nishant-d) [reply pkg/gitops/GitOpsConfigService.go:155]: in secret all data should be base64 encoded, either we have to encode or library will encode

File: pkg/gitops/GitOpsConfigService.go (hunk 1 above)
Comment (nishant-d) [reply pkg/gitops/GitOpsConfigService.go]: any gitops related call will fail in this window

Comment (vikramdevtron) [reply pkg/gitops/GitOpsConfigService.go]: > any gitops related call will fail in this window

let's make atleast one active. will add a validation for this.

File: internal/util/K8sUtil.go (hunk 2 above)
Comment (vikramdevtron) [reply internal/util/K8sUtil.go:103]: > then please remove it.

done

File: internal/util/K8sUtil.go (hunk 3 above)
Comment (vikramdevtron) [reply internal/util/K8sUtil.go:117]: done

File: pkg/gitops/GitOpsConfigService.go (hunk 4 above)
Comment (vikramdevtron) [reply pkg/gitops/GitOpsConfigService.go:155]: > in secret all data should be base64 encoded, either we have to encode or library will encode

okay, making changes around this.

**Pull request merged**